docs/infrastructure/administration.md

```diff
@@ -0,0 +1,393 @@
+credentials as the master production instance.
+
+[mysql workbench]: https://www.mysql.com/products/workbench/
+[aws-rds-cert]: https://github.com/mozilla/treeherder/blob/master/deployment/aws/rds-combined-ca-bundle.pem
+[rds ssl docs]: https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_MySQL.html#MySQL.Concepts.SSLSupport
+
+### Resetting dev/stage to a recent production snapshot
+
+The `treeherder-dev` and `treeherder-stage` RDS instances do not have regular user/sheriff
+interactions that provide certain sets of data (for example classifications of failures).
+
+As such it can be useful to periodically reset them using the latest daily production snapshot.
+This has to be performed by an admin of the `moz-devservices` AWS account, and can be requested
+by filing a bug similar to [bug 1469837].
+
+After an instance is reset:
+
+- The password for the MySQL user `th_admin` must be updated to match the one in `DATABASE_URL`
+  for that Heroku app (since the production password will have been inherited from the snapshot,
+  which won't match). See [Changing MySQL passwords].
+- [Perform a deployment] of the Heroku app, which will ensure that the Django migration and
+  code state are in sync (in case `master` has new migrations not yet seen on production).
+- [Reset the app's Redis cache]
+
+[bug 1469837]: https://bugzilla.mozilla.org/show_bug.cgi?id=1469837
+[changing mysql passwords]: #changing-mysql-passwords
+[perform a deployment]: #deploying-treeherder
+[reset the app's redis cache]: #resetting-the-redis-cache
+
+### Creating a temporary instance
+
+To create a new RDS instance based on the latest daily production DB snapshot:
+
+1. Ensure a bug is filed, which will act as a reminder to delete the instance later.
+2. Go to the [us-east-1 RDS snapshots page].
+3. Select the most recent `treeherder-prod` snapshot.
+4. From the "Actions" menu choose "Restore snapshot".
+5. Set the following options (leaving others at their defaults):
+   - DB instance class: `db.m4.xlarge` (same as dev/stage)
+   - DB instance identifier: `treeherder-dev-bug<NNNNNN>`
+   - Virtual private cloud: `treeherder-prod-vpc`
+   - Subnet group: `treeherder-dbgrp` (press the VPC refresh button to get this to appear)
+   - Public accessibility: Yes
+   - DB parameter group: `treeherder-mysql57`
+6. Select "Restore DB Instance".
+7. Wait 15-20 minutes for the instance to be created and report status as "Available".
+8. Select the instance, click "Modify", enter a randomly generated 30+ character password in
+   the "New master password" field, and press "Continue" to apply the change.
+9. Make a note of the DB hostname recorded under `Endpoint`. It will be in the form
+   `treeherder-dev-bugNNNNNN.HASH.us-east-1.rds.amazonaws.com`.
+
+To use the new instance locally, run this inside the Vagrant shell:
+
+```bash
+export DATABASE_URL='mysql://th_admin:PASSWORD@HOSTNAME/treeherder'
```

Comment:
I went through these steps and everything seemed to work ok, except I wasn't able to actually access the instance through either MySQLWorkbench or in Vagrant.  Not sure if you can spot anything that's missing here.  If not, I know we're out of time.  So I'd be happy to merge this as-is and I'll follow up with ops to figure out what else I might need to do to get it working.  I had Sarah try it, too, on my instance.  But she hit the same issue.  She's going to try creating her own instance now.

Reply:
I tried also - I set the Security Group to `treeherder_heroku-sg` instead of `default` since that's what the other instances are set to and I saw access denied messages in the instance log (selecting default, which is what Cam's instance is set to, gives me the same django OperationalError  "Can't connect to MySQL server on...").

Reply:
@camd there is no "Reset master credentials" event shown in the AWS console (unlike Sarah's instance). Did the page submission at the reset password step succeed, or was there a JS exception and the page got stuck, like seen in the recent AWS support ticket (which was CCed to treeherder-internal@)? They said they were going to fix that, sigh. Perhaps worth reopening the ticket? (You both have access - follow the links in the email thread.)

@sarah-clements yeah the security group needs setting manually, since it's not inherited from the snapshot either. I think this was on the previous version of my steps, however must have gotten lost in the rewrite.

After changing the security group on Cameron's instance I'm able to connect successfully (using the prod password, since the master password reset hadn't taken place on that instance). Sarah, re the access denied you saw - was TLS definitely configured? Or perhaps the master password change was still taking place?

I've updated this section to mention the security group change and using the classic AWS UI as a fallback for JS exceptions.

Reply:
I tried it again and I do seem to have access now, however I'm not receiving any data after trying several endpoints.

Reply:
> however I'm not receiving any data after trying several endpoints.

Has `DATABASE_URL` definitely been updated? (missing out the `export` will mean the environment variable won't be used by later commands). Run `env` in the shell to check its current value.

Reply:
Yup, it's been updated. Wondering whether @camd receives data with his instance?

Reply:
@sarah-clements meant to ask before, what do you mean by receive data?
The database isn't hooked up to anything, so won't be receiving data.
The existing data it already has should be accessible via the REST API using that DB however.
This might be what you mean, but just checking?

Reply:
I can access this instance now, using the prod password.  Thanks for the updates on this, Ed.  :)

Reply:
Ah, sorry I am getting data after all. I had a cached `failures` url and didn't realize it was out of our 4 month date range :)

Reply:
Since mine is working now, I'll delete it.